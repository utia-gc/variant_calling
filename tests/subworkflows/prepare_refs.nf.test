nextflow_workflow {

    name "Test Workflow Prepare_Refs"
    script "subworkflows/prepare_refs.nf"
    workflow "Prepare_Refs"

    test("Prepare_Refs successfully unzips and emits GZIP-compressed genome and GTF files.") {
        // instantiate helper objects
        def refs = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = file("${refs.getGenomeGzip()}")
                input[1] = file("${refs.getGTFGzip()}")
                """
            }
        }

        then {
            // test workflow success
            assert workflow.success
            assert workflow.trace.succeeded().size() == 4

            // test genome decompressed
            assert workflow.out.genome
            with(workflow.out.genome) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa/
            }

            // test genome fasta indexed
            assert workflow.out.genome_index
            with(workflow.out.genome_index) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa.fai/
            }

            // test sequence dictionary file produced
            with(workflow.out.genome_dict) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I\.dict/
            }

            // test annotations decompressed
            assert workflow.out.annotations
            with(workflow.out.annotations) {
                assert size() == 1
                assert get(0) ==~ /^.*\/annotations_I.gtf/
            }

            // test reproducibility
            assert snapshot(
                workflow.out.genome,
                workflow.out.genome_index,
                workflow.out.annotations,
            ).match()
        }


    }
    test("Prepare_Refs successfully emits uncompressed genome and GTF files when uncompressed inputs are given.") {
        // instantiate helper objects
        def refs = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = file("${refs.getGenome()}")
                input[1] = file("${refs.getGTF()}")
                """
            }
        }

        then {
            // test workflow success
            assert workflow.success
            assert workflow.trace.succeeded().size() == 2

            // test genome decompressed
            assert workflow.out.genome
            with(workflow.out.genome) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa/
            }

            // test genome fasta indexed
            assert workflow.out.genome_index
            with(workflow.out.genome_index) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa.fai/
            }

            // test sequence dictionary file produced
            with(workflow.out.genome_dict) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I\.dict/
            }

            // test annotations decompressed
            assert workflow.out.annotations
            with(workflow.out.annotations) {
                assert size() == 1
                assert get(0) ==~ /^.*\/annotations_I.gtf/
            }

            // test reproducibility
            assert snapshot(
                workflow.out.genome,
                workflow.out.genome_index,
                workflow.out.annotations,
            ).match()
        }

    }

}
