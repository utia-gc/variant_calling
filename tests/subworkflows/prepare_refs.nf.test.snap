{
    "Prepare_Refs successfully unzips and emits GZIP-compressed fasta and GTF files.": {
        "content": [
            {
                "0": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ],
                "1": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "annotations": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "fasta": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ]
            }
        ],
        "timestamp": "2023-09-27T15:44:42+0000"
    },
    "successfully emits uncompressed fasta and GTF files when uncompressed inputs are given.": {
        "content": [
            {
                "0": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ],
                "1": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "annotations": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "fasta": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ]
            }
        ],
        "timestamp": "2023-09-27T15:44:42+0000"
    },
    "successfully unzips and emits GZIP-compressed fasta and GTF files.": {
        "content": [
            {
                "0": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ],
                "1": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "annotations": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "fasta": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ]
            }
        ],
        "timestamp": "2023-09-27T15:44:42+0000"
    },
    "Prepare_Refs successfully unzips and emits GZIP-compressed genome and GTF files.": {
        "content": [
            [
                "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
            ],
            [
                "genome_I.fa.fai:md5,2cd76d936cbfa386b14154506c2041b2"
            ],
            [
                "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
            ]
        ],
        "timestamp": "2023-11-07T14:57:29.749656762"
    },
    "Prepare_Refs successfully emits uncompressed fasta and GTF files when uncompressed inputs are given.": {
        "content": [
            {
                "0": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ],
                "1": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "annotations": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "fasta": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ]
            }
        ],
        "timestamp": "2023-09-27T15:44:42+0000"
    },
    "Prepare_Refs successfully emits uncompressed genome and GTF files when uncompressed inputs are given.": {
        "content": [
            [
                "/utia-gc/ngs-test/raw/ngs/data/references/R64-1-1/genome_I.fa"
            ],
            [
                "genome_I.fa.fai:md5,2cd76d936cbfa386b14154506c2041b2"
            ],
            [
                "/utia-gc/ngs-test/raw/ngs/data/references/R64-1-1/annotations_I.gtf"
            ]
        ],
        "timestamp": "2023-11-07T14:57:40.865596797"
    }
}